BaseProject/src/de/tum/in/dbpra/model/dao/StageDAO.java

```diff
@@ -23,28 +23,78 @@ public void getPersonalStages(StageListBean stagelist, String firstname,
 
-		PreparedStatement pstmt = con.prepareStatement(query);
-		
-		if(firstname.equals("")||firstname.equals(null)){
-			throw new SQLException("Please enter a first name");
-		}
+		PreparedStatement pstmt1 = con.prepareStatement(query1);
+		pstmt1.setString(1, firstname);
+		pstmt1.setString(2, lastname);
+
+		ResultSet rs1 = pstmt1.executeQuery();
+		rs1.next();
+		int number = rs1.getInt("count");
 		
-		if(lastname.equals("")||lastname.equals(null)){
-			throw new SQLException("Please enter a last name");
+		if (number == 0){
+			con.rollback();
+			rs1.close();
+			pstmt1.close();
+			throw new VisitorNotFoundException("error1");
+		} else if (number > 1) {
+			con.rollback();
+			rs1.close();
+			pstmt1.close();
+			throw new SeveralVisitorsFouncException("error2");
+		} else {
+			PreparedStatement pstmt2 = con.prepareStatement(query2);
+			pstmt2.setString(1, firstname);
+			pstmt2.setString(2, lastname);
+			ResultSet rs2 = pstmt2.executeQuery();
+
+			while (rs2.next()) {
+				StageBean stage = new StageBean();
+				stage.setPersID(rs2.getString("personID"));
+				stage.setPersFirstName(rs2.getString("firstname"));
+				stage.setPersLastName(rs2.getString("lastname"));
+				stage.setStageID(rs2.getInt("stagenr"));
+				stage.setName(rs2.getString("stagename"));
+				stage.setSize(rs2.getDouble("size"));
+				stage.setBandName(rs2.getString("bandName"));
+				stage.setPerformanceStart(rs2.getTimestamp("performanceStart"));
+				stage.setPerformanceEnd(rs2.getTimestamp("performanceEnd"));
+				stagelist.setChild(stage);
+			}
+			rs2.close();
+			pstmt2.close();
+			
 		}
-		
+
+		con.commit();
+		rs1.close();
+		pstmt1.close();
+		con.close();
+
+	}
+	
+	public void getPersonalStagesByID(StageListBean stagelist, String firstname, String lastname, int ticketID) throws SQLException, ClassNotFoundException, VisitorNotFoundException, SeveralVisitorsFouncException {
+
+
+		String query = "SELECT DISTINCT person.personID, person.firstname, person.lastname, stage.stageNr, stage.stagename, stage.size, band.bandName, band.performanceStart, band.performanceEnd"
+				+ " FROM stage INNER JOIN lineup"
+				+ " ON stage.stageNr = lineup.stageNr INNER JOIN band ON lineup.bandID = band.bandID"
+				+ " INNER JOIN timetable ON timetable.bandID = band.bandID INNER JOIN visitor"
+				+ " ON timetable.visitorID = visitor.visitorID INNER JOIN Person ON visitor.visitorID = person.personID"
+				+ " WHERE person.firstname = ? AND person.lastname = ? AND visitor.ticketid = ?"
+				+ " ORDER BY person.personID, stage.stageNr, band.performanceStart;";
+
+		Connection con = getConnection();
+
+		PreparedStatement pstmt = con.prepareStatement(query);
 		pstmt.setString(1, firstname);
 		pstmt.setString(2, lastname);
+		pstmt.setInt(3, ticketID);
 
 		ResultSet rs = pstmt.executeQuery();
 		
-		if(!rs.next()){
-			throw new SQLException("There is no visitor " + firstname + " " +lastname);
-		}
-
-		while (rs.next()) {
+		if (rs.next()) {
```

Comment:
This must be a while loop, otherwise only the first element of the resultset will be displayed.